lib/rspec/matchers/built_in/has.rb

```diff
@@ -68,38 +70,87 @@ def private_predicate?
           end
         end
 
-        def predicate_exists?
-          @actual.respond_to? predicate
-        end
-
         if RSpec::Support::RubyFeatures.kw_args_supported?
           binding.eval(<<-CODE, __FILE__, __LINE__)
-          def predicate_matches?
+          def predicate_result
             if @kwargs.empty?
-              @actual.__send__(predicate, *@args, &@block)
+              @predicate_result = actual.__send__(predicate_method_name, *@args, &@block)
             else
-              @actual.__send__(predicate, *@args, **@kwargs, &@block)
+              @predicate_result = actual.__send__(predicate_method_name, *@args, **@kwargs, &@block)
             end
           end
           CODE
         else
-          def predicate_matches?
-            @actual.__send__(predicate, *@args, &@block)
+          def predicate_result
+            @predicate_result = actual.__send__(predicate_method_name, *@args, &@block)
           end
         end
 
-        def predicate
+        def predicate_method_name
+          predicate
+        end
```

Comment:
I feel like this should be in the has class as it's only used there? (I've checked the other PR so I think thats the case?)

Reply:
Ah actually, I see, its not used elsewhere in has, ignore this comment.